### PR TITLE
[main] Replace generated dependencies on java-headless with java

### PR DESCRIPTION
**Pull request by thcrain-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Our current RPM build of `msopenjdk-11` does not provide the `java-headless` name. This PR patches the `javapackages-tools` package to generate RPM dependencies on `java` rather than `java-headless` accordingly.

###### Change Log  

- Add patch to replace generated dependency on "java-headless" with "java"
- Amend epoch patch to fix expected test results
- Remove obsoletes statements that don't apply to Mariner
- Install test requirements with pip during check section

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Ongoing pipeline build

### CODE REVIEW

**Review by jslobodzian — COMMENTED** [review]
Please also fix these SPEC files:

	SPECS-EXTENDED/bsh/bsh.spec
	SPECS-EXTENDED/flute/flute.spec
	SPECS-EXTENDED/jakarta-commons-httpclient/jakarta-commons-httpclient.spec
	SPECS-EXTENDED/jing-trang/jing-trang.spec
	SPECS-EXTENDED/jss/jss.spec
	SPECS-EXTENDED/lasso/lasso.spec
	SPECS-EXTENDED/libvirt-java/libvirt-java.spec
	SPECS-EXTENDED/openmpi/openmpi.spec
	SPECS-EXTENDED/sblim-cim-client2/sblim-cim-client2.spec
	SPECS-EXTENDED/tomcat/tomcat.spec
	SPECS-EXTENDED/tomcatjss/tomcatjss.spec

**Comment by thcrain-msft** [comment]
> Please also fix these SPEC files:
> 
> ```
> SPECS-EXTENDED/bsh/bsh.spec
> SPECS-EXTENDED/flute/flute.spec
> SPECS-EXTENDED/jakarta-commons-httpclient/jakarta-commons-httpclient.spec
> SPECS-EXTENDED/jing-trang/jing-trang.spec
> SPECS-EXTENDED/jss/jss.spec
> SPECS-EXTENDED/lasso/lasso.spec
> SPECS-EXTENDED/libvirt-java/libvirt-java.spec
> SPECS-EXTENDED/openmpi/openmpi.spec
> SPECS-EXTENDED/sblim-cim-client2/sblim-cim-client2.spec
> SPECS-EXTENDED/tomcat/tomcat.spec
> SPECS-EXTENDED/tomcatjss/tomcatjss.spec
> ```

I'd like to do this in a separate PR, if that's alright.

**Review by jslobodzian — APPROVED** [review]

File: SPECS-EXTENDED/libvirt-java/libvirt-java.spec [hunk 1]
```diff
@@ -4,7 +4,7 @@ Summary:    Java bindings for the libvirt virtualization API
 Name:       libvirt-java
 Version:    0.4.9
 Prefix:     libvirt
-Release:    17%{?dist}
+Release:    18%{?dist}
```
Comment (PawelWMS) [inline SPECS-EXTENDED/libvirt-java/libvirt-java.spec:7]: May I ask you to revert the changes in this spec? I'm already migrating this file to `SPECS` [here](https://github.com/microsoft/CBL-Mariner/pull/1845) and adding some additional changes to it anyway.

File: SPECS-EXTENDED/bsh/bsh.spec [hunk 2]
```diff
@@ -60,7 +60,7 @@ BuildRequires:  ImageMagick
 BuildRequires:  desktop-file-utils
 %endif
 
-Requires:       java-headless
```
Comment (PawelWMS) [inline SPECS-EXTENDED/bsh/bsh.spec:63]: I think I recall this issue being mentioned. Do you know why the team providing the Java packages cannot add the `java-headless` subpackage? Do we just want to avoid the delay with such request?

Comment (PawelWMS) [reply SPECS-EXTENDED/bsh/bsh.spec:63]: An idea for an alternative solution: create our own dummy `java-headless` package, which `Requires: java`.

File: SPECS-EXTENDED/libvirt-java/libvirt-java.spec (hunk 1 above)
Comment (thcrain-msft) [reply SPECS-EXTENDED/libvirt-java/libvirt-java.spec]: Done, thanks!

File: SPECS-EXTENDED/bsh/bsh.spec (hunk 2 above)
Comment (thcrain-msft) [reply SPECS-EXTENDED/bsh/bsh.spec:63]: Our Java folks have already approved a request for the appropriate "Provides" to be added to the `msopenjdk-11` RPM. We just want to avoid the delay, since the change will come as part of their update cycle (ETA later this month). As for converting the manual `java-headless` dependencies, I think it's fine in this case. Any spec manually requiring `java` or `java-headless` should really be converted to use the `javapackages-tools` package to generate that requirement.
